app/scripts/modules/core/securityGroup/securityGroupReader.service.spec.ts

```diff
@@ -0,0 +1,197 @@
+import {mock} from 'angular';
+
+import {API_SERVICE, Api} from 'core/api/api.service';
+import {SECURITY_GROUP_TRANSFORMER_SERVICE, SecurityGroupTransformerService} from './securityGroupTransformer.service';
+import {
+  SECURITY_GROUP_READER_SERVICE, SecurityGroupReaderService,
+  ISecurityGroupDetail
+} from './securityGroupReader.service';
+import {ISecurityGroup} from 'core/domain';
+import {APPLICATION_MODEL_BUILDER, ApplicationModelBuilder} from 'core/application/applicationModel.builder';
+import {Application} from 'core/application/application.model';
+
+describe('Service: securityGroupReader', function () {
+
+  let $q: ng.IQService,
+    $http: ng.IHttpBackendService,
+    $scope: ng.IRootScopeService,
+    API: Api,
+    applicationModelBuilder: ApplicationModelBuilder,
+    reader: SecurityGroupReaderService;
+
+  beforeEach(mock.module(APPLICATION_MODEL_BUILDER, SECURITY_GROUP_TRANSFORMER_SERVICE, SECURITY_GROUP_READER_SERVICE, API_SERVICE));
+  beforeEach(
+    mock.inject(function (_$q_: ng.IQService,
+                          $httpBackend: ng.IHttpBackendService,
+                          $rootScope: ng.IRootScopeService,
+                          _API_: Api,
+                          _applicationModelBuilder_: ApplicationModelBuilder,
+                          _serviceDelegate_: any,
+                          securityGroupTransformer: SecurityGroupTransformerService,
+                          _securityGroupReader_: SecurityGroupReaderService) {
+      reader = _securityGroupReader_;
+      $http = $httpBackend;
+      API = _API_;
+      applicationModelBuilder = _applicationModelBuilder_;
+      $q = _$q_;
+      $scope = $rootScope.$new();
+
+      spyOn(securityGroupTransformer, 'normalizeSecurityGroup')
+        .and
+        .callFake((securityGroup: ISecurityGroup) => {
+          return $q.when(securityGroup);
+        });
+      spyOn(_serviceDelegate_, 'getDelegate').and.returnValue(
+        {
+          resolveIndexedSecurityGroup: (idx: any, container: ISecurityGroup, id: string) => {
+            return idx[container.account][container.region][id];
+          }
+        }
+      );
+    })
+  );
+
+  it('should inject defined objects', function () {
+    expect($q).toBeDefined();
+    expect($http).toBeDefined();
+    expect($scope).toBeDefined();
+    expect(API).toBeDefined();
+    expect(reader).toBeDefined();
+  });
```

Comment:
we have this in our test cases in a bunch of other places.  i'll make a todo to remove because you're right.  the test will barf if it can't be injected.